Remove `env_logger::init` within `InitBuilder` allowing other global subscribers to be set

apekros opened the pull request:
#213

Maximkaaa reviewed (changes requested):
Please, see my comment in the related issue. And also, this should work the same way in native and wasm environments.

apekros commented:
> Please, see my comment in the related issue. And also, this should work the same way in native and wasm environments.

I've hopefully addressed those comments with my changes now
